cvat/apps/engine/task.py

```diff
@@ -481,11 +481,25 @@ def _create_task_manifest_from_cloud_data(
     db_storage: models.CloudStorage,
     sorted_media: List[str],
     manifest: ImageManifestManager,
+    *,
+    stop_frame: int,
+    start_frame: int = 0,
+    step: int = 1,
     dimension: models.DimensionType = models.DimensionType.DIM_2D,
 ) -> None:
     cloud_storage_instance = db_storage_to_storage_instance(db_storage)
-    content = cloud_storage_instance.bulk_download_to_memory(sorted_media)
-    manifest.link(sources=content, DIM_3D=dimension == models.DimensionType.DIM_3D)
+    index_range = range(start_frame, stop_frame, step)
+    filtered_sorted_media = [sorted_media[idx] for idx in index_range]
+    content_generator = cloud_storage_instance.bulk_download_to_memory(filtered_sorted_media)
+
+    manifest.link(
+        sources=content_generator,
+        DIM_3D=dimension == models.DimensionType.DIM_3D,
+        stop=stop_frame,
+        start=start_frame,
+        step=step,
+        are_sources_filtered=True,
```

Comment:
Can this parameter be removed? E.g. if the `sources` contain something like a list of paths or bytes or the resulting structure (which is now just a dict).

Reply:
Initially, I would like to add handling not only cases when the `sources` variable is generator returned after partial downloading of images but also add the ability to create manifest from images generator without passing stop frame. But I've removed handling this case since we do not use it anywhere.